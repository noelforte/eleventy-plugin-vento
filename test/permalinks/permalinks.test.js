import { runBuild } from '../get-instance.js';
import { expect, test } from 'vitest';

const results = await runBuild(import.meta.dirname);

test('render a page at a nested permalink', async () => {
	const page = results.find(({ url }) => url === '/page/at/nested/permalink/');
	expect(page.content).toMatchSnapshot();
});

test('render a page at root url', async () => {
	const page = results.find(({ url }) => url === '/');
	expect(page.content).toMatchSnapshot();
});

test('render a page at a dynamic permalink', async () => {
	const page = results.find(
		({ url }) => url === '/directory-data-test-page.html'
	);

	expect(page).toBeDefined();
});

test('skips rendering a file', () => {
	const writtenTemplates = results.filter(({ url }) => Boolean(url));
	expect(writtenTemplates.length).toMatchSnapshot();
});
