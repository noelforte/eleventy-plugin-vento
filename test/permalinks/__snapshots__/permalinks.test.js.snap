// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`render a page at a nested permalink 1`] = `
"<!doctype html>
<html>
	<head><title>Nested Page</title></head>
	<body>
		<header>
			<h1>Page title: Nested Page</h1>
			</header>
		<main>
This content lives at \`/page/at/nested/permalink/\`
</main>
		<footer>
			<p>Page footer</p>
		</footer>
	</body>
</html>
"
`;

exports[`render a page at root url 1`] = `
"<!doctype html>
<html>
	<head><title>Root page</title></head>
	<body>
		<header>
			<h1>Page title: Root page</h1>
			</header>
		<main>
This should render on the homepage.
</main>
		<footer>
			<p>Page footer</p>
		</footer>
	</body>
</html>
"
`;

exports[`skips rendering a file 1`] = `3`;
